test/feature-benchmark/scenarios.py

```diff
@@ -1424,6 +1424,153 @@ def benchmark(self) -> MeasurementSource:
         )
 
 
+class MySqlCdc(Scenario):
+    pass
+
+
+class MySqlInitialLoad(MySqlCdc):
+    """Measure the time it takes to read 1M existing records from MySQL
+    when creating a materialized source"""
+
+    def shared(self) -> Action:
+        return TdAction(
+            f"""
+$ postgres-execute connection=postgres://mz_system:materialize@${{testdrive.materialize-internal-sql-addr}}
+ALTER SYSTEM SET enable_mysql_source = true
+
+$ mysql-connect name=mysql url=mysql://root@mysql password=${{arg.mysql-root-password}}
+
+$ mysql-execute name=mysql
+DROP DATABASE IF EXISTS public;
+CREATE DATABASE public;
+USE public;
+
+SET @i:=0;
+CREATE TABLE pk_table (pk BIGINT PRIMARY KEY, f2 BIGINT);
+INSERT INTO pk_table SELECT @i:=@i+1, @i*@i FROM mysql.time_zone t1, mysql.time_zone t2 LIMIT {self.n()};
+"""
+        )
+
+    def before(self) -> Action:
+        return TdAction(
+            """
+> DROP SOURCE IF EXISTS mz_source_mysqlcdc;
+> DROP CLUSTER IF EXISTS source_cluster CASCADE
+            """
+        )
+
+    def benchmark(self) -> MeasurementSource:
+        return Td(
+            f"""
+$ postgres-execute connection=postgres://mz_system:materialize@${{testdrive.materialize-internal-sql-addr}}
+ALTER SYSTEM SET enable_mysql_source = true
```

Comment:
@def-: Do you understand why I need to do this again in `benchmark()` with https://github.com/MaterializeInc/materialize/pull/25052/commits/5a1ba0edb09d693e5bb184d2bf27a24087a1f590? I already set it in `shared()`; I don't need to repeat it in the other scenario `MySqlStreaming` in `before()` where I also specify it in `shared()`.

Reply:
1. I would enable_mysql_source in all of CI in the __init__.py we have somewhere (I keep forgetting the path)
2. `shared` is run only once, but you start up another Mz instance later, which didn't have `shared` run. It's meant for things like initializing Kafka. But everything that has to be run against each Mz instance shouldn't be in `shared`. I don't understand why it isn't required in MySqlStreaming, maybe because another benchmark happens to run before and enable it?